### ISSUE TITLE
Add support for default map iterator yielding (key,value) pairs

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#24733]
While working on updating CLBG codes for Chapel 2.0, one of the rewrites was to change:

```chapel
for (k,v) in freqs.items() do ...
```

into:

```chapel
for (k,v) in zip(freqs.keys(), freqs.values()) do ...
```

in order to avoid the deprecation warning for the `items()` iterator.  This issue captures the desire to reinstate such a `(key,value)`-yielding iterator on maps, and to make it the default iterator for maps as proposed in https://github.com/chapel-lang/chapel/issues/14718 to permit the above to be written as:

```chapel
for (k,v) in freqs do ...
```

My understanding is that an aspect of the language or implementation relating to tuples and their ref-ness vs. copy behavior is what led us to deprecate these, and I understand that, but I think we're taking a big productivity hit in this area as a result.

**Comment by mppf** [comment]
> My understanding is that an aspect of the language or implementation relating to tuples and their ref-ness vs. copy behavior is what led us to deprecate these, and I understand that, but I think we're taking a big productivity hit in this area as a result.

Right, and one idea proposed in #14718 is to allow explicit control of the yield intent of tuple elements, as with e.g. `proc map.these(): (const ref keyType, ref valType)`
